File: backend/app/models/lead_enquiry.py
import enum

from sqlalchemy import Column, Integer, String, DateTime, Enum, func

from app.core.database import Base


class PriorityEnum(str, enum.Enum):
    low = "Low"
    medium = "Medium"
    high = "High"


class LeadEnquiry(Base):
    __tablename__ = "lead_enquiry"

    id = Column(Integer, primary_key=True, index=True)

    product_code = Column(String, nullable=False)
    product_name = Column(String, nullable=False)
    product_description = Column(String, nullable=True)

    customer_name = Column(String, nullable=False)
    customer_phone = Column(String, nullable=False)

    salesman_code = Column(String, nullable=False)
    salesman_name = Column(String, nullable=False)

    location_code = Column(String, nullable=False)
    location_name = Column(String, nullable=False)

    priority = Column(Enum(PriorityEnum, name="priority_enum"), nullable=False)

    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: frontend/src/types/leadEnquiry.ts
export type Priority = "Low" | "Medium" | "High";

export interface LeadEnquiryFormData {
  product_code: string;
  product_name: string;
  product_description: string;
  customer_name: string;
  customer_phone: string;
  salesman_code: string;
  salesman_name: string;
  location_code: string;
  location_name: string;
  priority: Priority;
}

export interface LeadEnquiry extends LeadEnquiryFormData {
  id: number;
  created_at: string;
}

export const emptyLeadEnquiryForm: LeadEnquiryFormData = {
  product_code: "",
  product_name: "",
  product_description: "",
  customer_name: "",
  customer_phone: "",
  salesman_code: "",
  salesman_name: "",
  location_code: "",
  location_name: "",
  priority: "Low",
};
